### PR TITLE
feat(publish-nightly): publishing nightly che-plugin-registry to gh-pages

### PR DESCRIPTION
**Pull request by sunix**
Signed-off-by: Sun Tan <sutan@redhat.com>

### What does this PR do?
Publish the nightly plugin registry to gh-pages (in addition of surge.sh)

### What issues does this PR fix or reference?
Related to https://github.com/eclipse/che/issues/19268

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: .github/workflows/publish-content-push-main.yml [hunk 1]
```diff
@@ -50,3 +50,11 @@ jobs:
            export DEPLOY_DOMAIN=https://che-plugin-registry-main.surge.sh
            echo "DEPLOY_DOMAIN=$DEPLOY_DOMAIN" >> $GITHUB_ENV
            surge ./content --domain $DEPLOY_DOMAIN
+      - name: Publish to gh-pages
+        env:
+          GITHUB_TOKEN: ${{ secrets.CHE_BOT_GITHUB_TOKEN }}
+          VERSION_DIR: nightly
```
Comment (benoitf) [inline .github/workflows/publish-content-push-main.yml:56]: is it nightly as we push on every commit ?

Comment (sunix) [reply .github/workflows/publish-content-push-main.yml:56]: it is taking the nightly tag of the docker image, isn't it ?

File: .github/workflows/publish-content-push-main.yml [hunk 2]
```diff
@@ -50,3 +50,11 @@ jobs:
...
+        run: |
+          export BUILD_DIR="$PWD/content"
+          export GH_PAGES_DIR="$PWD/gh-pages"
+          ./build/dev//publish-devfile-registry-to-gh-pages.sh
```
Comment (benoitf) [inline .github/workflows/publish-content-push-main.yml:60]: looks the filename is somehow wrong ?

**Review by ericwill — APPROVED** [review]

File: .github/workflows/publish-content-push-main.yml (hunk 1 above)
Comment (benoitf) [reply .github/workflows/publish-content-push-main.yml:56]: it's taking image being build after each commit (image may be named nightly which is wrong at it's updated at every commit)
so `current`, `next`, `main` (like main branch) would be better name

Comment (sunix) [reply .github/workflows/publish-content-push-main.yml]: Updated using main

File: .github/workflows/publish-content-push-main.yml (hunk 2 above)
Comment (sunix) [reply .github/workflows/publish-content-push-main.yml]: fixed

**Review by benoitf — APPROVED** [review]

**Pull request merged**